### PR TITLE
Add batch_fc op in contrib

### PR DESCRIPTION
**Pull request by ForFishes**
This Op can calculate batch fc. Notice: It currently supports GPU device. This Op exists in contrib, which means that it is not shown to the public. 
Part dim of batch_fc_op:
X.dim = slot_pairs_num * ins_num * in_dim
W.dim = slot_pairs_num * in_dim * out_dim
b.dim = slot_pairs_num * out_dim
output.dim = slot_pairs_num * ins_num * out_dim

### CODE REVIEW

**Review by gavin1332 — COMMENTED** [review]
Does bias add and activation calculated separately?

**Review by gavin1332 — COMMENTED** [review]
LGTM

**Review by lanxianghit — DISMISSED** [review]

File: python/paddle/fluid/contrib/layers/nn.py [hunk 1]
```diff
@@ -1298,3 +1298,65 @@ def rank_attention(input,
         attrs={"MaxRank": max_rank,
                "MaxSize": max_size})
     return output
+
+
+def batch_fc(input, param_size, param_attr, bias_size, bias_attr, act=None):
+    """
+    **Batch FC layer**
+    This Op can calculate BatcheFC.
```
Comment (Boyan-Liu) [inline python/paddle/fluid/contrib/layers/nn.py]: BatcheFC是拼写错误？

Comment (ForFishes) [reply python/paddle/fluid/contrib/layers/nn.py]: OK， i will fix it.

Comment (Boyan-Liu) [inline python/paddle/fluid/contrib/layers/nn.py]: 也请解释下具体释义，目前不能直观理解op功能

Comment (ForFishes) [reply python/paddle/fluid/contrib/layers/nn.py]: OK, i will fix it,thanks~

**Review by Boyan-Liu — APPROVED** [review]

**Review by lanxianghit — APPROVED** [review]

**Pull request merged**